src/browser/views.py

```diff
@@ -318,19 +341,32 @@ def repo_settings(request, repo_base, repo):
     returns the settings page for a repo.
     '''
     username = request.user.get_username()
-    manager = DataHubManager(user=username, repo_base=repo_base)
-    collaborators = manager.list_collaborators(repo)
 
-    # remove the current user from the collaborator list
-    collaborators = [c.get('username') for c in collaborators]
-    collaborators = filter(lambda x: x != '' and x !=
-                           username, collaborators)
+    with custom_manager_error_handler(username, repo_base, repo):
+        manager = DataHubManager(user=username, repo_base=repo_base)
+        collaborators = manager.list_collaborators(repo)
+
+    # if the public role is in collaborators, note that it's already added
+    repo_is_public = False
+    for collaborator in collaborators:
+        if collaborator['username'] == settings.PUBLIC_ROLE:
+            repo_is_public = True
+
+    # remove the current user, public user from the collaborator list
+    # collaborators = [c.get('username') for c in collaborators]
+    collaborators = filter(
+        lambda x: x['username'] != '' and x['username'] != username and
+        x['username'] != settings.PUBLIC_ROLE, collaborators)
+
+    public_role = settings.PUBLIC_ROLE
```

Comment:
If you move `public_role = settings.PUBLIC_ROLE` up to line 350 or so, you can save repeating settings.PUBLIC_ROLE a few times.